lib/src/protocol/models/resource.dart

```diff
@@ -61,17 +64,28 @@ abstract class Resource {
   String? signature;
 
   static Future<Resource> parse(String payload) async {
-    final jsonMessage = jsonDecode(payload);
-    // TODO(ethan-tbd): validate jsonMessage against resource
+    final jsonResource = jsonDecode(payload) as Map<String, dynamic>?;
+    if (jsonResource == null) {
+      throw Exception('payload is not a valid JSON object');
+    }
+    Validator.validate(jsonResource, 'resource');
+
+    final jsonResourceData = jsonResource['data'];
+    final resourceMetadata = jsonResource['metadata'] as ResourceMetadata?;
+    if (resourceMetadata == null) {
+      throw Exception('metadata property is required');
+    }
+    final resourceKind = resourceMetadata.kind.name;
+    Validator.validate(jsonResourceData, resourceKind);
```

Comment:
This logic seems duplicated, would it be better to put it in a class that can be parsed with `data` and `metadata`?

Reply:
will make sure to cover this in https://github.com/TBD54566975/tbdex-dart/issues/6!